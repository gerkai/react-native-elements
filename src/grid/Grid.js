import React, { Component, PropTypes } from 'react';
import { View, StyleSheet, TouchableOpacity } from 'react-native';
import Row from './Row';

class Grid extends Component {
  static propTypes = {
    containerStyle: PropTypes.any,
    onPress: PropTypes.func,
    activeOpacity: PropTypes.number,
  }

  static defaultProps = {
    activeOpacity: 1,
  }

  styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: this.isRow() ? 'column' : 'row',
    },
    opacityContainer: {
      flex: 1,
    },
  })

  isRow() {
    let isRow = false;
    React.Children.forEach(this.props.children, (child) => {
      if (child && child.type === Row) {
        isRow = true;
      }
    });

    return isRow;
  }

  render() {
    const {onPress, activeOpacity, containerStyle} = this.props;

    if (onPress) {
      return (
        <TouchableOpacity style={opacityContainer} activeOpacity={activeOpacity} onPress={onPress}>
          <View
            style={[
              this.styles.container,
              containerStyle && containerStyle,
            ]}
            {...this.props}
          >
            {this.props.children}
          </View>
        </TouchableOpacity>
      );
    }

    return (
      <View
        style={[
          this.styles.container,
          containerStyle && containerStyle,
        ]}
        {...this.props}
      >
        {this.props.children}
      </View>
    );
  }
}

export default Grid;
