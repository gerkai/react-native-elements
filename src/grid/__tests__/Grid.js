import React from 'react';
import {Text} from 'react-native';
import {shallow} from 'enzyme';
import Grid from '../Grid';

describe('Grid Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Grid />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('should render children', () => {
    const component = shallow(<Grid><Text>Hi</Text></Grid>);

    expect(component.find('Text').length).toBe(1);
  });
});
