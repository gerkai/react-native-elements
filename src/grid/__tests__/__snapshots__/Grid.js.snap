// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Component should render without issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    activeOpacity={1}
    style={
        Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
            },
            undefined,
          ]
    }
/>,
  "nodes": Array [
    <View
      activeOpacity={1}
      style={
            Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  },
                  undefined,
                ]
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Grid
        activeOpacity={1}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Grid {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "activeOpacity": 1,
        },
        "refs": Object {},
        "state": null,
        "styles": Object {
          "container": Object {
            "flex": 1,
            "flexDirection": "row",
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          activeOpacity={1}
          style={
                    Array [
                              Object {
                                "flex": 1,
                                "flexDirection": "row",
                              },
                              undefined,
                            ]
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <View
          activeOpacity={1}
          style={
                    Array [
                              Object {
                                "flex": 1,
                                "flexDirection": "row",
                              },
                              undefined,
                            ]
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Grid
    activeOpacity={1}
/>,
}
`;
